core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -655,6 +655,15 @@ object SparkSubmit {
           // scalastyle:on println
         }
         System.exit(CLASS_NOT_FOUND_EXIT_STATUS)
+      case e: NoClassDefFoundError =>
+        e.printStackTrace(printStream)
+        if (e.getMessage.contains("org/apache/hadoop/hive")) {
+          // scalastyle:off println
+          printStream.println(s"Failed to load hive class.")
+          printStream.println("You need to build Spark with -Phive and -Phive-thriftserver.")
+          // scalastyle:on println
+        }
+        System.exit(CLASS_NOT_FOUND_EXIT_STATUS)
```

Comment:
This issue happens only when SPARK_PREPEND_CLASSES is true. In this case the spark assembly jar doesn't contain SparkSQLCLIDriver because spark is built without -Phive-thriftserver, but it can still be found in CLASSPATH because SPARK_PREPEND_CLASSES is true. But hive related class can not be found since it is not in the spark assembly jar.